integration-tests/typescript-angular/src/app/todo-lists.yaml/client.service.ts

```diff
@@ -36,19 +48,21 @@ export class ApiClient {
     )
   }
 
-  private _queryParams(
-    queryParams: Record<
-      string,
-      boolean | number | string | string[] | undefined | null
-    >
-  ): HttpParams {
-    const result = new HttpParams()
-    Object.entries(queryParams).forEach(([name, value]) => {
-      if (value !== undefined && value !== null) {
-        result.set(name, String(value))
+  private _queryParams(queryParams: QueryParams): HttpParams {
+    return Object.entries(queryParams).reduce((result, [name, value]) => {
+      if (
+        typeof value === "string" ||
+        typeof value === "boolean" ||
+        typeof value === "number"
+      ) {
+        return result.set(name, value)
+      } else if (value === null || value === undefined) {
+        return result
       }
-    })
-    return result
+      throw new Error(
```

Comment:
Getting the behavior right for array and object types is tricky. Throw an exception for now to allow the `QueryParams` type to still accept them to make the stripe code happy and come back to it later